### DISCUSSION TITLE
[Share] Fix Line Endings in Abstract

### DISCUSSION THREAD

**Discussion by windingwind** [windingwind/zotero-actions-tags#163] [Action Scripts]
### Description

See https://github.com/windingwind/zotero-actions-tags/issues/162

A right-click menu option to remove the line endings in the abstract of items.
Something similar to the Abstract Cleaner plugin: https://github.com/dcartertod/zotero-plugins

Some items have abstracts that are wrongly formatted, with many line endings remaining in the abstract.
For some items, no abstract is imported at all. If I copy from Adobe Acrobat, the line endings are also copied. I still need to remove them automatically.
That functionality was available in the Abstract Cleaner plugin. But it is not compatible with Zotero 7.

### Event

None

### Operation

Script

### Data

```javascript
/**
 * Remove the line endings in the abstract of items.
 * @author windingwind
 * @usage Right-click menu
 * @link https://github.com/windingwind/zotero-actions-tags/discussions/163
 * @see https://github.com/windingwind/zotero-actions-tags/discussions/163
 */
if (!item) {
	return;
}
// https://github.com/dcartertod/zotero-plugins/blob/main/ZoteroAbstractCleaner/chrome/content/ZoteroAbstractCleaner.js
var abstract = item.getField('abstractNote');
// bit more tedious but preserves paragraphs
abstract = abstract.replace(/\r/gm," ");
abstract = abstract.replace(/\n\n/gm,"<br>");
abstract = abstract.replace(/\n/gm," ");
abstract = abstract.replace(/<br>/gm,"\n\n");
abstract = abstract.replace(/ +/g," ");
item.setField('abstractNote',abstract);
```

**Comment by mjthoraval** [comment]
Thank you very much!
